### PR TITLE
HTTPS update on files in library folder

### LINKED ISSUE #2247: Replace HTTP links with HTTPS when possible
There are quite a few http links throughout the codebase. Some have to stay that way, but many can be shifted to https. 

Good example is [this file](https://github.com/openemr/openemr/blob/master/library/globals.inc.php).
- [This line](https://github.com/openemr/openemr/blob/c5649fd9a9dc61b5bf1908f9cafd00c0cb426512/library/globals.inc.php#L616) has to stay http. If it's shifted to https, the link breaks.
- However [this line](https://github.com/openemr/openemr/blob/c5649fd9a9dc61b5bf1908f9cafd00c0cb426512/library/globals.inc.php#L392) can be shifted to https. 

TASK:
- Use an IDE such as PHPStorm to load the code
- Search for all use cases of `http` in [this folder](https://github.com/openemr/openemr/tree/master/library
) to start
- Go through each of them and see if they can be replaced with https
- Make a pull request!

### PR DESCRIPTION
**Pull request by marieram**
All lines with http have been changed to https. This fixes #2247.

### CODE REVIEW

File: library/ajax/execute_background_services.php [hunk 1]
```diff
@@ -37,10 +37,10 @@
  * Returns a count of due messages for current user.
  *
  * @package   OpenEMR
- * @link      http://www.open-emr.org
- * @author    EMR Direct <http://www.emrdirect.com/>
+ * @link      s://www.open-emr.org
```
Comment (bradymiller) [inline library/ajax/execute_background_services.php]: missing a couple letters here

**Comment by bradymiller** [comment]
hi @marieram , Only found one issue; otherwise looks good. After the one issue is addressed, then will be ready for codebase :)

File: library/ajax/execute_background_services.php (hunk 1 above)
Comment (marieram) [reply library/ajax/execute_background_services.php]: Thanks, @bradymiller. I found a few more http. I will revise this and push the changes shortly.

Comment (marieram) [reply library/ajax/execute_background_services.php]: Revised and fixed.

File: library/classes/Filtreatment_class.php [hunk 2]
```diff
@@ -3,9 +3,9 @@
  * FILTREATMENT CLASS FILE
  *
  *
- * @author Cristian Năvălici {@link http://www.lemonsoftware.eu} lemonsoftware [at] gmail [.] com
+ * @author Cristian Năvălici {@link https://www.lemonsoftware.eu} lemonsoftware [at] gmail [.] com
```
Comment (bradymiller) [inline library/classes/Filtreatment_class.php]: This link only works with http , so needs to stay as http

File: library/classes/html2text.class.php [hunk 3]
```diff
@@ -34,30 +34,30 @@
 /**
  *  Takes HTML and converts it to formatted, plain text.
  *
- *  Thanks to Alexander Krug (http://www.krugar.de/) to pointing out and
+ *  Thanks to Alexander Krug (https://www.krugar.de/) to pointing out and
```
Comment (bradymiller) [inline library/classes/html2text.class.php]: This link only works with http , so needs to stay as http

File: library/classes/html2text.class.php [hunk 4]
```diff
@@ -34,30 +34,30 @@
...
+ *  Thanks to Darius Kasperavicius (https://www.dar.dar.lt/) for
  *  suggesting the addition of $allowed_tags and its supporting function
  *  (which I slightly modified). Updated 3/12/04.
  *
  *  Thanks to Justin Dearing for pointing out that a replacement for the
  *  <TH> tag was missing, and suggesting an appropriate fix.
  *  Updated 8/25/04.
  *
- *  Thanks to Mathieu Collas (http://www.myefarm.com/) for finding a
+ *  Thanks to Mathieu Collas (https://www.myefarm.com/) for finding a
```
Comment (bradymiller) [inline library/classes/html2text.class.php]: This link only works with http , so needs to stay as http

File: library/classes/html2text.class.php [hunk 5]
```diff
@@ -67,29 +67,29 @@
...
- *  Thanks to Marc Bertrand (http://www.dresdensky.com/) for
+ *  Thanks to Marc Bertrand (https://www.dresdensky.com/) for
  *  suggesting a revision to the word wrapping functionality; if you
  *  specify a $width of 0 or less, word wrapping will be ignored.
  *  Updated 11/02/06.
  *
  *  *** Big housecleaning updates below:
  *
- *  Thanks to Colin Brown (http://www.sparkdriver.co.uk/) for
+ *  Thanks to Colin Brown (https://www.sparkdriver.co.uk/) for
```
Comment (bradymiller) [inline library/classes/html2text.class.php]: https://www.sparkdriver.co.uk/

Comment (bradymiller) [reply library/classes/html2text.class.php]: This link only works with http , so needs to stay as http

File: library/classes/html2text.class.php [hunk 6]
```diff
@@ -67,29 +67,29 @@
...
  *  suggesting the fix to handle </li> and blank lines (whitespace).
- *  Christian Basedau (http://www.movetheweb.de/) also suggested the
+ *  Christian Basedau (https://www.movetheweb.de/) also suggested the
  *  blank lines fix.
  *
- *  Special thanks to Marcus Bointon (http://www.synchromedia.co.uk/),
- *  Christian Basedau, Norbert Laposa (http://ln5.co.uk/),
+ *  Special thanks to Marcus Bointon (https://www.synchromedia.co.uk/),
+ *  Christian Basedau, Norbert Laposa (https://ln5.co.uk/),
```
Comment (bradymiller) [inline library/classes/html2text.class.php]: This link (https://ln5.co.uk/) only works with http , so needs to stay as http

Comment (marieram) [reply library/classes/html2text.class.php]: 👍

File: library/classes/html2text.class.php (hunk 4 above)
Comment (marieram) [reply library/classes/html2text.class.php]: 👍

File: library/classes/html2text.class.php (hunk 3 above)
Comment (marieram) [reply library/classes/html2text.class.php]: 👍

File: library/classes/Filtreatment_class.php (hunk 2 above)
Comment (marieram) [reply library/classes/Filtreatment_class.php]: 👍

**Comment by bradymiller** [comment]
hi @marieram , Looks good except for a couple broken https site links (looking at those site was like seeing a blast from the past btw :) ). When those are addressed, then will bring this into the codebase. thanks, -brady

**Comment by bradymiller** [comment]
hi @marieram , Looks great. Going to bring this into codebase now. Thanks for the improvements! -brady

**Comment by danehrlich1** [comment]
@marieram thanks for your quick work on this issue. 

Is there anything else you’d like to work on / goals you have career wise?

We usually create custom issues for ppl to work on based on their skills and needs

**Comment by marieram** [comment]
@danehrlich1, I'm new to coding. If you guys have anything similar to this issue, I'd be happy to help. With guidance and instructions, I can take it on. Any custom code required, may be to advanced. I will keep an eye out for any open issues, especially those with the first-timer-only label. Thanks for asking!


> @marieram thanks for your quick work on this issue. Is there anything else you’d like to work on / goals you have career wise? We usually create custom issues for ppl to work on based on their skills and needs
> […](#)
> On Feb 27, 2019, at 8:51 PM, Brady Miller ***@***.***> wrote: hi @marieram

**Pull request merged**